apps/meteor/server/lib/rooms/roomCoordinator.ts

```diff
@@ -41,14 +42,29 @@ class RoomCoordinatorServer extends RoomCoordinator {
 				sender: AtLeast<IUser, '_id' | 'name' | 'username'>,
 				notificationMessage: string,
 				userId: string,
+				language?: string,
 			): Promise<{ title: string | undefined; text: string; name: string | undefined }> {
-				const title = `#${await this.roomName(room, userId)}`;
-				const useRealName = settings.get<boolean>('UI_Use_Real_Name');
-				const senderName = getUserDisplayName(sender.name, sender.username, useRealName);
+				const showPushMessage = settings.get<boolean>('Push_show_message');
+				const showUserOrRoomName = settings.get<boolean>('Push_show_username_room');
+				const lng = language || settings.get('Language') || 'en';
 
-				const text = `${senderName}: ${notificationMessage}`;
+				let text;
+				let title;
+				let name;
+				if (showPushMessage) {
+					const useRealName = settings.get<boolean>('UI_Use_Real_Name');
+					const senderName = getUserDisplayName(sender.name, sender.username, useRealName);
+					text = `${senderName}: ${notificationMessage}`;
+				} else {
+					text = i18n.t('You_have_a_new_message', { lng });
+				}
```

Comment:
This piece of code seems to be repeated over the room types, do you think we could extract it to a function and have it unit tested?